### ISSUE TITLE
SPAdes v3.15.0 crashes with --trusted-contigs option

### ISSUE THREAD

**Issue by fplaza** [ablab/spades#660]
Hi,

I am trying to perform metagenome assembly by combining Illumina and PacBio data.
SPAdes v3.15.0 crashes with "--trusted-contigs" option (assembly from PacBio data generated with metaFlye) but works fine with Illumina reads only.
[spades.log](https://github.com/ablab/spades/files/5927255/spades.log)
[params.txt](https://github.com/ablab/spades/files/5927256/params.txt)

Thanks in advance for your help,
Florian

**Comment by asl** [comment]
Tagging @andrewprzh @uncerso 
Can you share your data with us so we could reproduce and fix the issue?

Thanks!

**Comment by fplaza** [comment]
Unfortunately, I can not for legal reasons.

**Comment by asl** [comment]
Well, I'm afraid then there are not so many possibilities. Does the problem reproduce with ``-untrusted-contigs``?

**Comment by fplaza** [comment]
Hi,
SPAdes terminated successfully by switching from '--meta' to '--careful' mode

**Comment by asl** [comment]
I doubt this would bring meaningful results for a metagenome :)

**Comment by fplaza** [comment]
Sure, but I guess this information will help you for solving this bug.

**Comment by asl** [comment]
Unfortunately, it won't help.

**Comment by fplaza** [comment]
Combining '--meta' with '--untrusted-contigs' works as well. 
I will identify a public dataset were I can reproduce that bug.

**Comment by fplaza** [comment]
Hi,

Here is a public dataset with which you can reproduce the bug:
https://filesender.renater.fr/?s=download&token=a5abce0f-7524-412e-ad54-6c728df59c11

```
~/SPAdes-3.15.1-Linux/bin/spades.py \
-1 illumina_data_sub_20M_fastp/SRR8073716_1.fastq.gz \
-2 illumina_data_sub_20M_fastp/SRR8073716_2.fastq.gz \
--trusted-contigs metaflye_pacbio_assembly/assembly.fasta \
--meta \
-t $(nproc) \
-o pacbio_illumina_hybrid_assembly
```

**Comment by asl** [comment]
Tagging @uncerso @andrewprzh

**Comment by asl** [comment]
Reproduced, thanks!

**Comment by hoytpr** [comment]
Having same problem here using trusted or untrusted contigs.

**Comment by asl** [comment]
@hoytpr Your problem is completely unrelated as log reads:
```
FileNotFoundError: [Errno 2] No such file or directory: '/scratch/phoyt/deole/SPADES/LACaaabR1R2acad_hybrid/split_input/tmp/contigs'
```

Clearly an I/O issue on your scratch directory. You may want to talk with your system administrator about this issue

**Comment by hoytpr** [comment]
Thanks I'll check with them.

**Comment by asl** [comment]
This was fixed in SPAdes 3.15.3

**Comment by fplaza** [comment]
Hello
--trusted-contigs and untrusted-contigs options are not available anymore in metaSPAdes mode.
Is this the new expected behaviour?
